cinder/volume/drivers/lightos/lightos.py

```diff
@@ -834,9 +809,24 @@ def _wait_for_snapshot_state(
 
     def _wait_for_volume_deleted(self, project_name, timeout, vol_uuid):
         """Wait until the volume has been deleted."""
-        return self._wait_for_volume_state(
-            project_name, timeout, states=(
-                'Deleted', 'Deleting', 'UNKNOWN'), vol_uuid=vol_uuid)
+        assert vol_uuid, 'LightOS volume UUID must be specified'
+        states = ('Deleted', 'Deleting', 'UNKNOWN')
+
+        stop = time.time() + timeout
+        while time.time() <= stop:
+            (status_code,
+             resp) = self._get_lightos_volume(project_name,
+                                              timeout=self.logical_op_timeout,
+                                              vol_uuid=vol_uuid)
+            if status_code == httpstatus.NOT_FOUND:
+                return 'Deleted'
+            state = resp.get('state', 'UNKNOWN') if \
+                status_code == httpstatus.OK and resp else 'UNKNOWN'
+            if state in states:
```

Comment:
thnk this would be clearr if you just check state== deleting (deleted and unkown should never be returned)